astropy fails to compile with gcc 14

### Description

Attempteting to build from source with gcc 14 fails with the following erro:

```
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/wcslib_celprm_wrap.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/wcslib_celprm_wrap.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      astropy/wcs/src/wcslib_celprm_wrap.c: In function ‘PyCelprm_deepcopy’:
      astropy/wcs/src/wcslib_celprm_wrap.c:167:22: error: initialization of ‘PyCelprm *’ from incompatible pointer type ‘PyObject *’ {aka ‘struct _object *’} [-Wincompatible-pointer-types]
        167 |     PyCelprm* copy = PyCelprm_new(&PyCelprmType, NULL, NULL);
            |                      ^~~~~~~~~~~~
      In file included from /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/arrayobject.h:5,
                       from astropy/wcs/include/astropy_wcs/pyutil.h:16,
                       from astropy/wcs/include/astropy_wcs/wcslib_celprm_wrap.h:4,
                       from astropy/wcs/src/wcslib_celprm_wrap.c:3:
      astropy/wcs/src/wcslib_celprm_wrap.c: In function ‘PyCelprm_set_ref’:
      astropy/wcs/src/wcslib_celprm_wrap.c:327:25: error: passing argument 1 of ‘PyArray_DIMS’ from incompatible pointer type [-Wincompatible-pointer-types]
        327 |     size = PyArray_SIZE(value_array);
            |                         ^~~~~~~~~~~
            |                         |
            |                         PyObject * {aka struct _object *}
      /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarrayobject.h:79:59: note: in definition of macro ‘PyArray_SIZE’
         79 | #define PyArray_SIZE(m) PyArray_MultiplyList(PyArray_DIMS(m), PyArray_NDIM(m))
            |                                                           ^
      In file included from /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarrayobject.h:12:
      /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarraytypes.h:1520:35: note: expected ‘const PyArrayObject *’ {aka ‘const struct tagPyArrayObject *’} but argument is of type ‘PyObject *’ {aka ‘struct _object *’}
       1520 | PyArray_DIMS(const PyArrayObject *arr)
            |              ~~~~~~~~~~~~~~~~~~~~~^~~
      astropy/wcs/src/wcslib_celprm_wrap.c:327:25: error: passing argument 1 of ‘PyArray_NDIM’ from incompatible pointer type [-Wincompatible-pointer-types]
        327 |     size = PyArray_SIZE(value_array);
            |                         ^~~~~~~~~~~
            |                         |
            |                         PyObject * {aka struct _object *}
      /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarrayobject.h:79:76: note: in definition of macro ‘PyArray_SIZE’
         79 | #define PyArray_SIZE(m) PyArray_MultiplyList(PyArray_DIMS(m), PyArray_NDIM(m))
            |                                                                            ^
      /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarraytypes.h:1502:35: note: expected ‘const PyArrayObject *’ {aka ‘const struct tagPyArrayObject *’} but argument is of type ‘PyObject *’ {aka ‘struct _object *’}
       1502 | PyArray_NDIM(const PyArrayObject *arr)
            |              ~~~~~~~~~~~~~~~~~~~~~^~~
      astropy/wcs/src/wcslib_celprm_wrap.c:348:35: error: passing argument 1 of ‘PyArray_DATA’ from incompatible pointer type [-Wincompatible-pointer-types]
        348 |     data = (double*) PyArray_DATA(value_array);
            |                                   ^~~~~~~~~~~
            |                                   |
            |                                   PyObject * {aka struct _object *}
      /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarraytypes.h:1508:35: note: expected ‘const PyArrayObject *’ {aka ‘const struct tagPyArrayObject *’} but argument is of type ‘PyObject *’ {aka ‘struct _object *’}
       1508 | PyArray_DATA(const PyArrayObject *arr)
            |              ~~~~~~~~~~~~~~~~~~~~~^~~
      error: command '/lib/ccache/bin/gcc' failed with exit code 1
      [end of output]
```

The full build log (including creating a virtual env) below.  This is with python312 from Arch linux. See with both np1.x and np2.0.

<details>

```
~
21:13:58 [belanna] @ python -m venv /tmp/astropy
~
✔ 21:14:05 [belanna] @ vox activate /tmp/astropy/
Activated "/tmp/astropy/".

(astropy) ~
✔ 21:14:12 [belanna] @ pip install --no-binary astropy astropy
Collecting astropy
  Downloading astropy-6.1.0.tar.gz (7.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.0/7.0 MB 11.9 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting numpy>=1.23 (from astropy)
  Using cached numpy-1.26.4-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)
Collecting pyerfa>=2.0.1.1 (from astropy)
  Downloading pyerfa-2.0.1.4-cp39-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.7 kB)
Collecting astropy-iers-data>=0.2024.4.29.0.28.48 (from astropy)
  Using cached astropy_iers_data-0.2024.5.13.0.30.12-py3-none-any.whl.metadata (5.1 kB)
Collecting PyYAML>=3.13 (from astropy)
  Using cached PyYAML-6.0.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)
Collecting packaging>=19.0 (from astropy)
  Using cached packaging-24.0-py3-none-any.whl.metadata (3.2 kB)
Using cached astropy_iers_data-0.2024.5.13.0.30.12-py3-none-any.whl (1.9 MB)
Using cached numpy-1.26.4-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.0 MB)
Using cached packaging-24.0-py3-none-any.whl (53 kB)
Downloading pyerfa-2.0.1.4-cp39-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (738 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 738.7/738.7 kB 11.0 MB/s eta 0:00:00
Using cached PyYAML-6.0.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (724 kB)
Building wheels for collected packages: astropy
  Building wheel for astropy (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for astropy (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [1603 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-312
      creating build/lib.linux-x86_64-cpython-312/astropy
      .....
      SNIPPED A LOT OF LINES ABOUT COPYING FILES
      ......
      Compiling astropy/convolution/_convolve.pyx because it changed.
      [1/1] Cythonizing astropy/convolution/_convolve.pyx
      building 'astropy.convolution._convolve' extension
      creating build/temp.linux-x86_64-cpython-312
      creating build/temp.linux-x86_64-cpython-312/astropy
      creating build/temp.linux-x86_64-cpython-312/astropy/convolution
      creating build/temp.linux-x86_64-cpython-312/astropy/convolution/src
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/convolution/_convolve.c -o build/temp.linux-x86_64-cpython-312/astropy/convolution/_convolve.o -UNDEBUG -fPIC
      In file included from astropy/convolution/_convolve.c:1271:
      astropy/convolution/src/convolve.h:88:19: warning: inline function ‘convolve3d’ declared but never defined
         88 | FORCE_INLINE void convolve3d(DTYPE * const result,
            |                   ^~~~~~~~~~
      astropy/convolution/src/convolve.h:74:19: warning: inline function ‘convolve2d’ declared but never defined
         74 | FORCE_INLINE void convolve2d(DTYPE * const result,
            |                   ^~~~~~~~~~
      astropy/convolution/src/convolve.h:60:19: warning: inline function ‘convolve1d’ declared but never defined
         60 | FORCE_INLINE void convolve1d(DTYPE * const result,
            |                   ^~~~~~~~~~
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/convolution/src/convolve.c -o build/temp.linux-x86_64-cpython-312/astropy/convolution/src/convolve.o -UNDEBUG -fPIC
      gcc -shared -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto build/temp.linux-x86_64-cpython-312/astropy/convolution/_convolve.o build/temp.linux-x86_64-cpython-312/astropy/convolution/src/convolve.o -L/usr/lib -o build/lib.linux-x86_64-cpython-312/astropy/convolution/_convolve.cpython-312-x86_64-linux-gnu.so
      building 'astropy.stats._fast_sigma_clip' extension
      creating build/temp.linux-x86_64-cpython-312/astropy/stats
      creating build/temp.linux-x86_64-cpython-312/astropy/stats/src
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/stats/src/compute_bounds.c -o build/temp.linux-x86_64-cpython-312/astropy/stats/src/compute_bounds.o
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/stats/src/fast_sigma_clip.c -o build/temp.linux-x86_64-cpython-312/astropy/stats/src/fast_sigma_clip.o
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/stats/src/wirth_select.c -o build/temp.linux-x86_64-cpython-312/astropy/stats/src/wirth_select.o
      gcc -shared -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto build/temp.linux-x86_64-cpython-312/astropy/stats/src/compute_bounds.o build/temp.linux-x86_64-cpython-312/astropy/stats/src/fast_sigma_clip.o build/temp.linux-x86_64-cpython-312/astropy/stats/src/wirth_select.o -L/usr/lib -o build/lib.linux-x86_64-cpython-312/astropy/stats/_fast_sigma_clip.cpython-312-x86_64-linux-gnu.so
      Compiling astropy/stats/_stats.pyx because it changed.
      [1/1] Cythonizing astropy/stats/_stats.pyx
      building 'astropy.stats._stats' extension
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/stats/_stats.c -o build/temp.linux-x86_64-cpython-312/astropy/stats/_stats.o
      gcc -shared -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto build/temp.linux-x86_64-cpython-312/astropy/stats/_stats.o -L/usr/lib -o build/lib.linux-x86_64-cpython-312/astropy/stats/_stats.cpython-312-x86_64-linux-gnu.so
      Compiling astropy/table/_np_utils.pyx because it changed.
      [1/1] Cythonizing astropy/table/_np_utils.pyx
      building 'astropy.table._np_utils' extension
      creating build/temp.linux-x86_64-cpython-312/astropy/table
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/table/_np_utils.c -o build/temp.linux-x86_64-cpython-312/astropy/table/_np_utils.o
      gcc -shared -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto build/temp.linux-x86_64-cpython-312/astropy/table/_np_utils.o -L/usr/lib -o build/lib.linux-x86_64-cpython-312/astropy/table/_np_utils.cpython-312-x86_64-linux-gnu.so
      Compiling astropy/table/_column_mixins.pyx because it changed.
      [1/1] Cythonizing astropy/table/_column_mixins.pyx
      building 'astropy.table._column_mixins' extension
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/table/_column_mixins.c -o build/temp.linux-x86_64-cpython-312/astropy/table/_column_mixins.o
      gcc -shared -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto build/temp.linux-x86_64-cpython-312/astropy/table/_column_mixins.o -L/usr/lib -o build/lib.linux-x86_64-cpython-312/astropy/table/_column_mixins.cpython-312-x86_64-linux-gnu.so
      building 'astropy.time._parse_times' extension
      creating build/temp.linux-x86_64-cpython-312/astropy/time
      creating build/temp.linux-x86_64-cpython-312/astropy/time/src
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/time/src/parse_times.c -o build/temp.linux-x86_64-cpython-312/astropy/time/src/parse_times.o
      gcc -shared -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto build/temp.linux-x86_64-cpython-312/astropy/time/src/parse_times.o -L/usr/lib -o build/lib.linux-x86_64-cpython-312/astropy/time/_parse_times.cpython-312-x86_64-linux-gnu.so
      building 'astropy.utils._compiler' extension
      creating build/temp.linux-x86_64-cpython-312/astropy/utils
      creating build/temp.linux-x86_64-cpython-312/astropy/utils/src
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/utils/src/compiler.c -o build/temp.linux-x86_64-cpython-312/astropy/utils/src/compiler.o
      gcc -shared -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-z,pack-relative-relocs -flto=auto build/temp.linux-x86_64-cpython-312/astropy/utils/src/compiler.o -L/usr/lib -o build/lib.linux-x86_64-cpython-312/astropy/utils/_compiler.cpython-312-x86_64-linux-gnu.so
      building 'astropy.wcs._wcs' extension
      creating build/temp.linux-x86_64-cpython-312/astropy/wcs
      creating build/temp.linux-x86_64-cpython-312/astropy/wcs/src
      creating build/temp.linux-x86_64-cpython-312/cextern
      creating build/temp.linux-x86_64-cpython-312/cextern/wcslib
      creating build/temp.linux-x86_64-cpython-312/cextern/wcslib/C
      creating build/temp.linux-x86_64-cpython-312/cextern/wcslib/C/flexed
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/astropy_wcs.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/astropy_wcs.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/astropy_wcs_api.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/astropy_wcs_api.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/distortion.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/distortion.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/distortion_wrap.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/distortion_wrap.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/docstrings.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/docstrings.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/pipeline.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/pipeline.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/pyutil.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/pyutil.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      astropy/wcs/src/pyutil.c: In function ‘set_pscards’:
      astropy/wcs/src/pyutil.c:807:5: warning: ‘strncpy’ specified bound 72 equals destination size [-Wstringop-truncation]
        807 |     strncpy((*ps)[i].value, strvalue, 72);
            |     ^
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/sip.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/sip.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/sip_wrap.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/sip_wrap.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/str_list_proxy.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/str_list_proxy.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/unit_list_proxy.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/unit_list_proxy.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/util.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/util.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/wcslib_auxprm_wrap.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/wcslib_auxprm_wrap.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      gcc -fno-strict-overflow -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -DNPY_TARGET_VERSION=NPY_1_23_API_VERSION -UDEBUG -I/tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include -Icextern/wcslib/C -Iastropy/wcs/include -I/tmp/astropy/include -I/usr/include/python3.12 -c astropy/wcs/src/wcslib_celprm_wrap.c -o build/temp.linux-x86_64-cpython-312/astropy/wcs/src/wcslib_celprm_wrap.o -std=gnu99 -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
      astropy/wcs/src/wcslib_celprm_wrap.c: In function ‘PyCelprm_deepcopy’:
      astropy/wcs/src/wcslib_celprm_wrap.c:167:22: error: initialization of ‘PyCelprm *’ from incompatible pointer type ‘PyObject *’ {aka ‘struct _object *’} [-Wincompatible-pointer-types]
        167 |     PyCelprm* copy = PyCelprm_new(&PyCelprmType, NULL, NULL);
            |                      ^~~~~~~~~~~~
      In file included from /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/arrayobject.h:5,
                       from astropy/wcs/include/astropy_wcs/pyutil.h:16,
                       from astropy/wcs/include/astropy_wcs/wcslib_celprm_wrap.h:4,
                       from astropy/wcs/src/wcslib_celprm_wrap.c:3:
      astropy/wcs/src/wcslib_celprm_wrap.c: In function ‘PyCelprm_set_ref’:
      astropy/wcs/src/wcslib_celprm_wrap.c:327:25: error: passing argument 1 of ‘PyArray_DIMS’ from incompatible pointer type [-Wincompatible-pointer-types]
        327 |     size = PyArray_SIZE(value_array);
            |                         ^~~~~~~~~~~
            |                         |
            |                         PyObject * {aka struct _object *}
      /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarrayobject.h:79:59: note: in definition of macro ‘PyArray_SIZE’
         79 | #define PyArray_SIZE(m) PyArray_MultiplyList(PyArray_DIMS(m), PyArray_NDIM(m))
            |                                                           ^
      In file included from /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarrayobject.h:12:
      /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarraytypes.h:1520:35: note: expected ‘const PyArrayObject *’ {aka ‘const struct tagPyArrayObject *’} but argument is of type ‘PyObject *’ {aka ‘struct _object *’}
       1520 | PyArray_DIMS(const PyArrayObject *arr)
            |              ~~~~~~~~~~~~~~~~~~~~~^~~
      astropy/wcs/src/wcslib_celprm_wrap.c:327:25: error: passing argument 1 of ‘PyArray_NDIM’ from incompatible pointer type [-Wincompatible-pointer-types]
        327 |     size = PyArray_SIZE(value_array);
            |                         ^~~~~~~~~~~
            |                         |
            |                         PyObject * {aka struct _object *}
      /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarrayobject.h:79:76: note: in definition of macro ‘PyArray_SIZE’
         79 | #define PyArray_SIZE(m) PyArray_MultiplyList(PyArray_DIMS(m), PyArray_NDIM(m))
            |                                                                            ^
      /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarraytypes.h:1502:35: note: expected ‘const PyArrayObject *’ {aka ‘const struct tagPyArrayObject *’} but argument is of type ‘PyObject *’ {aka ‘struct _object *’}
       1502 | PyArray_NDIM(const PyArrayObject *arr)
            |              ~~~~~~~~~~~~~~~~~~~~~^~~
      astropy/wcs/src/wcslib_celprm_wrap.c:348:35: error: passing argument 1 of ‘PyArray_DATA’ from incompatible pointer type [-Wincompatible-pointer-types]
        348 |     data = (double*) PyArray_DATA(value_array);
            |                                   ^~~~~~~~~~~
            |                                   |
            |                                   PyObject * {aka struct _object *}
      /tmp/pip-build-env-aqqa9rix/overlay/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarraytypes.h:1508:35: note: expected ‘const PyArrayObject *’ {aka ‘const struct tagPyArrayObject *’} but argument is of type ‘PyObject *’ {aka ‘struct _object *’}
       1508 | PyArray_DATA(const PyArrayObject *arr)
            |              ~~~~~~~~~~~~~~~~~~~~~^~~
      error: command '/lib/ccache/bin/gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for astropy
Failed to build astropy
ERROR: Could not build wheels for astropy, which is required to install pyproject.toml-based projects
```


</details>

### Expected behavior

compile!

### How to Reproduce

1. install gcc 14
2. 
```bash
python -m venv /tmp/astropy
. /tmp/astropy/bin/activate
pip install --no-binary astropy astropy
```


### Versions

N/A because could not get astropy to install.  See both with latest release from pypi and the default branch.

```
gcc --version
gcc (GCC) 14.1.1 20240507
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/astropy/wcs/src/wcslib_celprm_wrap.c
+++ b/astropy/wcs/src/wcslib_celprm_wrap.c
@@ -164,7 +164,7 @@ static PyObject* PyCelprm_copy(PyCelprm* self)
 
 static PyObject* PyCelprm_deepcopy(PyCelprm* self)
 {
-    PyCelprm* copy = PyCelprm_new(&PyCelprmType, NULL, NULL);
+    PyCelprm* copy = (PyCelprm*) PyCelprm_new(&PyCelprmType, NULL, NULL);
     if (copy == NULL) return NULL;
 
     memcpy(copy->x, self->x, sizeof(struct celprm));
@@ -321,7 +321,7 @@ static int PyCelprm_set_ref(PyCelprm* self, PyObject* value, void* closure)
         return 0;
     }
 
-    PyObject* value_array = PyArray_ContiguousFromAny(value, NPY_DOUBLE, 1, 1);
+    PyArrayObject* value_array = (PyArrayObject*) PyArray_ContiguousFromAny(value, NPY_DOUBLE, 1, 1);
     if (!value_array) return -1;
 
     size = PyArray_SIZE(value_array);

--- a/astropy/wcs/src/wcslib_prjprm_wrap.c
+++ b/astropy/wcs/src/wcslib_prjprm_wrap.c
@@ -146,7 +146,7 @@ static PyObject* PyPrjprm_copy(PyPrjprm* self)
 
 static PyObject* PyPrjprm_deepcopy(PyPrjprm* self)
 {
-    PyPrjprm* copy = PyPrjprm_new(&PyPrjprmType, NULL, NULL);
+    PyPrjprm* copy = (PyPrjprm*) PyPrjprm_new(&PyPrjprmType, NULL, NULL);
     if (copy == NULL) return NULL;
 
     memcpy(copy->x, self->x, sizeof(struct prjprm));
@@ -510,11 +510,12 @@ static PyObject* PyPrjprm_get_pv(PyPrjprm* self, void* closure)
     int k;
     Py_ssize_t size = PVN;
     double *pv;
-    PyObject* pv_array;
-
+    PyObject* pv_pyobj;
+    PyArrayObject* pv_array;
     if (is_prj_null(self)) return NULL;
 
-    pv_array = (PyArrayObject*) PyArray_SimpleNew(1, &size, NPY_DOUBLE);
+    pv_pyobj = PyArray_SimpleNew(1, &size, NPY_DOUBLE);
+    pv_array = (PyArrayObject*) pv_pyobj;
     if (pv_array == NULL) return NULL;
     pv = (double*) PyArray_DATA(pv_array);
 
@@ -526,7 +527,7 @@ static PyObject* PyPrjprm_get_pv(PyPrjprm* self, void* closure)
         }
     }
 
-    return pv_array;
+    return pv_pyobj;
 }
 
 
@@ -535,7 +536,7 @@ static int PyPrjprm_set_pv(PyPrjprm* self, PyObject* value, void* closure)
     int k, modified;
     npy_intp size;
     double *data;
-    PyObject* value_array = NULL;
+    PyArrayObject* value_array = NULL;
     int skip[PVN];
 
     if (is_prj_null(self) || is_readonly(self)) return -1;
@@ -550,7 +551,7 @@ static int PyPrjprm_set_pv(PyPrjprm* self, PyObject* value, void* closure)
         return 0;
     }
 
-    value_array = PyArray_ContiguousFromAny(value, NPY_DOUBLE, 1, 1);
+    value_array = (PyArrayObject*) PyArray_ContiguousFromAny(value, NPY_DOUBLE, 1, 1);
     if (!value_array) return -1;
 
     size = PyArray_SIZE(value_array);
@@ -654,7 +655,8 @@ static PyObject* PyPrjprm_set_pvi(PyPrjprm* self, PyObject* args, PyObject* kwds
     PyObject* index = NULL;
     PyObject* value = NULL;
     PyObject* flt_value = NULL;
-    PyObject* value_array = NULL;
+    PyObject* value_array_pyobj = NULL;
+    PyArrayObject* value_array = NULL;
     const char* keywords[] = { "index", "value", NULL };
     PyArray_Descr* dbl_descr = PyArray_DescrNewFromType(NPY_DOUBLE);
 
@@ -705,9 +707,10 @@ static PyObject* PyPrjprm_set_pvi(PyPrjprm* self, PyObject* args, PyObject* kwds
         }
 
     } else {
-        if (PyArray_Converter(value, &value_array) == NPY_FAIL) {
+        if (PyArray_Converter(value, &value_array_pyobj) == NPY_FAIL) {
             return NULL;
         }
+        value_array = (PyArrayObject*) value_array_pyobj;
 
         size = PyArray_SIZE(value_array);
         if (size != 1) {
@@ -771,11 +774,11 @@ static PyObject* PyPrjprm_get_w(PyPrjprm* self, void* closure)
     Py_ssize_t size = 10;
     int k;
     double *w;
-    PyObject* w_array;
+    PyArrayObject* w_array;
 
     if (is_prj_null(self)) return NULL;
 
-    w_array = (PyObject*) PyArray_SimpleNew(1, &size, NPY_DOUBLE);
+    w_array = (PyArrayObject*) PyArray_SimpleNew(1, &size, NPY_DOUBLE);
     if (w_array == NULL) return NULL;
     w = (double*) PyArray_DATA(w_array);
 
@@ -787,7 +790,7 @@ static PyObject* PyPrjprm_get_w(PyPrjprm* self, void* closure)
         }
     }
 
-    return w_array;
+    return (PyObject*) w_array;
 }
 
 

--- a/docs/changes/wcs/16450.bugfix.rst
+++ b/docs/changes/wcs/16450.bugfix.rst
@@ -1,0 +1,1 @@
+Fix compilation with gcc 14, avoid implicit pointer conversions.

